examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,123 @@
+"""
+===============================================================================
+Silhouette analysis for sample data clustered using KMeans clustering algorithm
+===============================================================================
+
+Silhouette analysis can be used to study the separation distance between the
+resulting clusters. The silhouette plot displays a measure of how close each
+point in one cluster is to points in the neighboring clusters. This measure has
+a range of [-1, 1]. Silhoette coefficients (as these values are referred to as)
+near +1 indicate that the sample is far away from the neighboring clusters.
+A value of 0 indicates that the sample is on or very close to the decision 
+boundary between two neighboring clusters and negative values (upto -1) 
+indicate that those samples might have been assigned to the wrong cluster.
+"""
+
+from __future__ import print_function
+
+from sklearn.datasets import make_blobs
+from sklearn.cluster import KMeans
+from sklearn.metrics import silhouette_samples, silhouette_score
+
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+import numpy as np
+
+print(__doc__)
+
+# Generating the sample data from make_blobs
+# This particular setting has one distict cluster and 3 clusters placed close
+# together.
+X, y = make_blobs(n_samples=500,
+                  n_features=2,
+                  centers=4,
+                  cluster_std=1,
+                  center_box=(-10.0, 10.0),
+                  shuffle=True,
+                  random_state=1)  # For reproducibility
+
+range_n_clusters = [2, 4, 6]
+
+for n_clusters in range_n_clusters:
+    # Create a subplot with 1 row and 2 columns
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    fig.set_size_inches(18, 7)
+
+    # The 1st subplot is the silhouette plot
+    # The silhouette coefficient can range from -1, 1 but in this example all
+    # lie within [-0.1, 1]
+    ax1.set_xlim([-0.1, 1])
+    # The (n_clusters+1)*10 are the additional samples to demarcate the space
```

Comment:
Does `is for the additional space(s) in the y axis between the silhouette plots of the different clusters` look better?